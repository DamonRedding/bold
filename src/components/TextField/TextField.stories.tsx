import { action } from '@storybook/addon-actions'
import { boolean, text } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import React from 'react'

import { VFlow } from '../VFlow'

import { TextField } from './TextField'

storiesOf('Components|TextField', module)
  .add('default', () => (
    <TextField
      label={text('label', 'Text label')}
      placeholder='Nome'
      disabled={boolean('disabled', false)}
      onChange={action('changed')}
      error={text('error', '')}
      required={boolean('required', true)}
    />
  ))
  .add('password', () => (
    <TextField
      id='password'
      name='password'
      label='Password'
      placeholder='Type your password...'
      type='password'
      disabled={boolean('disabled', false)}
      onChange={action('changed')}
    />
  ))
  .add('with icon', () => (
    <VFlow>
      <TextField
        id='clickable'
        name='clickable'
        icon='zoomOutline'
        label='Clickable icon'
        onIconClick={action('icon-clicked')}
        disabled={boolean('disabled', false)}
        onChange={action('changed')}
      />
      <TextField
        id='decorative'
        name='decorative'
        icon='zoomOutline'
        iconPosition='left'
        label='Decorative icon'
        disabled={boolean('disabled', false)}
        onChange={action('changed')}
      />
    </VFlow>
  ))
