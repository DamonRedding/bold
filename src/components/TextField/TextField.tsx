import React, { useRef } from 'react'

import { randomStr } from '../../util/string'
import { FormControl } from '../FormControl'

import { TextInput, TextInputProps } from './TextInput'

export interface TextFieldProps extends TextInputProps {
  label?: string
  errorText?: string
}

export function TextField(props: TextFieldProps) {
  const { id, label, errorText, required, name, ...rest } = props

  const labelIdRef = useRef(`label-${randomStr()}`)
  const errorIdRef = useRef(`error-${randomStr()}`)

  const labelId = label && !id ? labelIdRef.current : undefined
  const errorId = errorText ? errorIdRef.current : undefined

  return (
    <FormControl
      label={label}
      htmlFor={id}
      labelId={labelId}
      errorId={errorId}
      error={errorText}
      required={required}
      data-name={name}
    >
      <TextInput
        id={id}
        name={name}
        required={required}
        aria-labelledby={labelId}
        invalid={!!errorText}
        aria-errormessage={errorId}
        {...rest}
      />
    </FormControl>
  )
}
