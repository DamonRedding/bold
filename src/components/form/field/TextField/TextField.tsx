import * as React from 'react'
import { WrappedFieldProps } from 'redux-form'
import { FormFieldProps } from '../../../form/FormField'
import { TextInput, TextInputProps } from '../../input/TextInput/TextInput'
import { Field, FieldProps } from '../../reduxForm/Field'

export interface TextFieldProps extends FormFieldProps, TextInputProps, Pick<FieldProps, 'normalize' | 'parse' | 'format'> {
    name: string
}

export class TextField extends React.Component<TextFieldProps> {

    render() {
        return (
            <Field
                {...this.props}
                render={this.renderInput}
            />
        )
    }

    private renderInput = (props: WrappedFieldProps) => (
        <TextInput
            {...this.props}
            {...props.input}
        />
    )

}
