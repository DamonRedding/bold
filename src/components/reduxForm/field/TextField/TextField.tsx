import * as React from 'react'
import { TextInputProps, TextInput } from '../../../input/TextInput'
import { Field, FieldProps } from '../../Field'
import { FormFieldProps } from '../../../form/FormField'
import { WrappedFieldProps } from 'redux-form'

export interface TextFieldProps extends FormFieldProps, TextInputProps, Pick<FieldProps, 'normalize' | 'parse' | 'format'> {
    name: string
}

export class TextField extends React.Component<TextFieldProps> {

    render() {
        return (
            <Field
                {...this.props}
                render={this.renderInput}
            />
        )
    }

    private renderInput = (props: WrappedFieldProps) => (
        <TextInput
            {...this.props}
            {...props.input}
        />
    )

}
