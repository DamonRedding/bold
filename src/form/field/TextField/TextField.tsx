import { TextInput, TextInputProps } from '../../../components/TextInput'
import { BaseFieldProps, withField } from '../../Field'

export type TextFieldProps = BaseFieldProps<TextInputProps, string>

export const TextField = withField<string, TextInputProps>(TextInput)
