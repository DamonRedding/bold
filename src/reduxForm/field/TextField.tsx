import * as React from 'react'
import { WrappedFieldProps } from 'redux-form'
import { TextInputProps, TextInput } from '../../components/input/TextInput'
import field, { FieldProps } from '../hoc/field'
import { FormFieldProps } from '../../components/form/FormField'

export interface TextFieldProps extends FormFieldProps, TextInputProps, Partial<WrappedFieldProps<any>> {
}

class TextFieldCmp extends React.Component<TextFieldProps> {

    render() {
        return (
            <TextInput
                {...this.props.input}
                placeholder={this.props.placeholder}
                maxLength={this.props.maxLength}
                disabled={this.props.disabled}
            />
        )
    }

}

export const TextField: React.ComponentClass<FieldProps & TextFieldProps> = field()(TextFieldCmp)
