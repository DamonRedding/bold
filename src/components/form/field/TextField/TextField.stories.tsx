import { boolean, withKnobs } from '@storybook/addon-knobs'
import { storiesOf } from '@storybook/react'
import * as React from 'react'

import { withForm } from '../../../../stories-addons/withForm'
import { withPropTypes } from '../../../../stories-addons/withPropTypes/index'
import { withTheme } from '../../../../stories-addons/withTheme'

import { TextField } from './TextField'

storiesOf('Form', module)
    .addDecorator(withPropTypes())
    .addDecorator(withKnobs)
    .addDecorator(withTheme())
    .addDecorator(withForm())
    .add('TextField', () => (
        <TextField
            disabled={boolean('disabled', false)}
            name='nome'
            label='Nome'
            placeholder='Nome'
            required
        />
    ))
    .add('TextField type=\'password\'', () =>
        <TextField name='senha' label='Senha' placeholder='Senha' type='password' required />
    )
