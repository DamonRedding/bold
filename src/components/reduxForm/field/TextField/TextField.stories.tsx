import * as React from 'react'

import { storiesOf } from '@storybook/react'
import { withKnobs } from '@storybook/addon-knobs'
import { TextField } from './TextField'
import { withPropTypes } from '../../../../stories-addons/withPropTypes/index'
import { withTheme } from '../../../../stories-addons/withTheme'
import { withForm } from '../../../../stories-addons/withForm'

storiesOf('TextField', module)
    .addDecorator(withPropTypes())
    .addDecorator(withKnobs)
    .addDecorator(withTheme())
    .addDecorator(withForm())
    .add('default', () =>
        <TextField name='nome' label='Nome' placeholder='Nome' />
    )
