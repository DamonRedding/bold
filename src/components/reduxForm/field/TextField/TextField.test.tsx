import * as React from 'react'
import { TextField } from '../TextField'
import { shallowRenderAndMatch, testForm } from '../../../../__tests__/test.defaults'

describe('TextField', () => {
    it('deve ser renderizado de forma correta', () => {
        shallowRenderAndMatch(testForm(() => <TextField name='test' placeholder='Test' maxLength={2} disabled={false} />))
    })
})
