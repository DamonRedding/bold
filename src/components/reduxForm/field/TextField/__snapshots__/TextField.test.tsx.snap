// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextField deve ser renderizado de forma correta 1`] = `<ReduxForm />`;
